plugins/modules/synchronize.py

```diff
@@ -548,10 +548,10 @@ def main():
             ssh_cmd_str = ' '.join(shlex_quote(arg) for arg in ssh_cmd)
             if ssh_args:
                 ssh_cmd_str += ' %s' % ssh_args
-            cmd.append('--rsh=%s' % ssh_cmd_str)
+            cmd.append(shlex_quote('--rsh=%s' % ssh_cmd_str))
 
     if rsync_path:
-        cmd.append('--rsync-path=%s' % rsync_path)
```

Comment:
Would it be possible to add tests for this change?

Reply:
There is already a test for it.
https://github.com/ansible-collections/ansible.posix/blob/main/tests/integration/targets/synchronize/tasks/main.yml#L154-L166

As @quidame mentioned later the integration tests were disabled sometime ago. I now enabled them. localy with docker it worked

```
TASK [synchronize : synchronize files using rsync_path (issue#7182)] ***********
changed: [testhost] => {"changed": true, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive '--rsync-path=sudo rsync' '--out-format=<<CHANGED>>%i %n%L' /root/ansible_collections/ansible/posix/tests/output/.tmp/output_dir/foo.txt /root/ansible_collections/ansible/posix/tests/output/.tmp/output_dir/foo.rsync_path", "msg": ">f+++++++++ foo.txt\n", "rc": 0, "stdout_lines": [">f+++++++++ foo.txt"]}

TASK [synchronize : assert] ****************************************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [synchronize : Cleanup] ***************************************************
changed: [testhost] => (item=foo.rsync_path) => {"ansible_loop_var": "item", "changed": true, "item": "foo.rsync_path", "path": "/root/ansible_collections/ansible/posix/tests/output/.tmp/output_dir/foo.rsync_path", "state": "absent"}
```

Reply:
Hi @Akasurde @flybyray,
Based on the code changes, I think we  probably need to append the integration test like below:
```YAML
- name: command-line options are quoted correctly (issue#24 and issue#190)
  synchronize:
    src: "{{output_dir}}/foo.txt"
    dest: "dummy:{{output_dir}}/foo.rsync_path_1"
  register: sync_result
  ignore_errors: true
- assert:
    that:
    - sync_result.cmd is search("'--rsh=.*'")
    - sync_result.cmd is search("'--out-format=.*'")
```